From 0a1c9776fcf4e6990db6be882a6df06f95ca5e56 Mon Sep 17 00:00:00 2001
From: Giovanni Ricca <giovanniricca@proton.me>
Date: Mon, 5 Jan 2026 19:29:20 +0100
Subject: [PATCH 3/3] sm8750-common: init: Set zram compression algorithm to
 lz4 for everyone

Offers nice decompression/compression speeds while being more
efficient than other algorithms (e.g. lzo or zstd).

Should have been the default for quite some time ...

Change-Id: I3be33c04bd5b150269d4f6ed4f4328d3d20fa580
---
 init/init.kernel.post_boot-memory.sh | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/init/init.kernel.post_boot-memory.sh b/init/init.kernel.post_boot-memory.sh
index 042dc4c..a6f4030 100755
--- a/init/init.kernel.post_boot-memory.sh
+++ b/init/init.kernel.post_boot-memory.sh
@@ -34,9 +34,6 @@ function configure_zram_parameters() {
 	MemTotalStr=`cat /proc/meminfo | grep MemTotal`
 	MemTotal=${MemTotalStr:16:8}
 
-	low_ram=`getprop ro.config.low_ram`
-
-
 	let RamSizeGB="( $MemTotal / 1048576 ) + 1"
 	diskSizeUnit=M
 	# Zram disk - 75%
@@ -47,10 +44,8 @@ function configure_zram_parameters() {
 		let zRamSizeMB=6144
 	fi
 
-	# And enable lz4 zram compression for Go targets.
-	if [ "$low_ram" == "true" ]; then
-		echo lz4 > /sys/block/zram0/comp_algorithm
-	fi
+	# And enable lz4 zram compression.
+	echo lz4 > /sys/block/zram0/comp_algorithm
 
 	if [ -f /sys/block/zram0/disksize ]; then
 		if [ -f /sys/block/zram0/use_dedup ]; then
-- 
2.55.0

